src/providers/idp/idp_auth_eval.c

```diff
@@ -0,0 +1,314 @@
+/*
+    SSSD
+
+    IdP Identity Backend Module - evalute authentication replies
+
+    Authors:
+        Sumit Bose <sbose@redhat.com>
+
+    Copyright (C) 2024 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <errno.h>
+#include <jansson.h>
+
+#include "util/util.h"
+#include "providers/idp/idp_auth.h"
+
+/* The following is a 1:1 copy from src/krb5_plugin/idp/idp_utils.c, let's see
+ * what would be the best way to make it public. */
```

Comment:
Why don't you just include the module in makefile?

Reply:
Hi,

because I want to use `sss_idp_oauth2_decode()` directly which is `static`. If you don't make I can make it public like `sss_idp_oauth2_decode_padata()`.

bye,
Sumit

Reply:
Hi,

in the new first two patches I made the need functions public.

bye,
Sumit